src/components/Board/Board.js

```diff
@@ -50,63 +46,71 @@ const Board = () => {
       player2Pos,
       p1StartPermission,
       p2StartPermission,
-      isSnake,
       snake,
-      isLadder,
       ladder,
       buttonAbility,
     },
     dispatch,
   ] = React.useReducer(reducer, initialStates);
 
   // I believe this should be in a useEffect since it is actually a side effect :)
+  //Done:)
+  //******************************************* QUESTION ********************************************
+  //Why is this considered a side effect? because the functionality depends on these two dependencies? or there's another reason?
+
   // Checking if any of the players reaches 100, to announce the game finished.
-  if (!gameOver && (player1Pos === 100 || player2Pos === 100)) {
-    dispatch({ type: "gameOver" });
-  }
+  React.useEffect(() => {
+    if (!gameOver && (player1Pos === 100 || player2Pos === 100)) {
+      dispatch({ type: "gameOver" });
+    }
+  }, [player1Pos, player2Pos, gameOver]);
 
-  // This function checks if the current player's position is considered a snake/ladder, if so, the position will be updated with the defined destination of that snake/ladder,
+  //This function checks if the current player's position is considered a snake/ladder, if so, the position will be updated with the defined destination of that snake/ladder,
   //Then, the snake/ladder state gets updated.
   const applySnakeOrLadder = (currentPlayerPos) => {
-    // foundSnake(currentPlayerPos) is being called multiple times,
-    // It would be worth saving it into a variable and re-use it :)
-    if (foundSnake(currentPlayerPos) !== undefined) {
-      // I can see that you understand how to use useReducer very well done!! :)
-      // I would also add that I think you can simplify most of your dispatch methods,
```

Comment:
Whenever you can, try to use early returns :)

We have this at the moment:
```js
if (foundSnake(currentPlayerPos) !== undefined) {
  // another nesting here with some code
}
```

We could do an early return and avoid nesting:
```js
if (foundSnake(currentPlayerPos) === undefined) return currentPlayerPos
// then you can add the other if statements below
```